app/javascript/controllers/patron_request_controller.js

```diff
@@ -8,9 +8,34 @@ export default class extends Controller {
     if (this.accordionTargets.length == 1) {
       this.removeAccordionStyling();
     }
+    this.nextStepButtonEventListeners();
     this.element.reset();
   }
 
+  emptyFields(accordion) {
+    const formData = new FormData(accordion.closest('form'));
+    return Array.from(accordion.querySelectorAll('[required], [data-required]')).find(x => formData.getAll(x.name).every(x => !x))
+  }
+
+  nextStepButtonEventListeners() {
+    var _this = this;
+    this.accordionTargets.forEach(accord => {
+      _this.enableNextButton(accord)
+      accord.addEventListener("input", function() {
+        _this.enableNextButton(accord)
+      });
```

Comment:
Can we add this event listener the stimulus way? (e.g.  something like`data-action="input->patronRequest#checkIfTheRequiredFieldsAreFilledOutAndEnableTheNextButton"`)

Reply:
Sounds good. Pushed